packages/plugins/rubygem-newt/rubygem-newt.spec

```diff
@@ -1,110 +1,81 @@
-BuildRequires: %{?scl_prefix_ruby}ruby(release)
-BuildRequires: %{?scl_prefix_ruby}ruby-devel
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
-Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
-# end specfile generated dependencies
-
-Requires: newt
+Requires: ruby(release)
+Requires: ruby
+Requires: ruby(rubygems)
+Requires: newt-devel
+
+BuildRequires: ruby(release)
+BuildRequires: ruby
+BuildRequires: ruby(rubygems)
 BuildRequires: newt-devel
+BuildRequires: ruby-devel
+BuildRequires: rubygems-devel
+# Compiler is required for build of gem binary extension.
+# https://fedoraproject.org/wiki/Packaging:C_and_C++#BuildRequires_and_Requires
+BuildRequires: gcc
+
+Provides: rubygem(%{gem_name}) = %{version}
 
 %description
 Ruby bindings for newt TUI library.
 
-
 %package doc
-Summary: Documentation for %{pkg_name}
+Summary: Documentation for %{name}
 Group: Documentation
-Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Requires: %{name} = %{version}-%{release}
 BuildArch: noarch
 
 %description doc
-Documentation for %{pkg_name}.
+Documentation for %{name}.
 
 %prep
-%{?scl:scl enable %{scl} - << \EOF}
-gem unpack %{SOURCE0}
-%{?scl:EOF}
-
-%setup -q -D -T -n  %{gem_name}-%{version}
-
-%{?scl:scl enable %{scl} - << \EOF}
-gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
-%{?scl:EOF}
+%setup -q -n  %{gem_name}-%{version}
 
 %build
 # Create the gem as gem install only works on a gem file
-%{?scl:scl enable %{scl} - << \EOF}
-gem build %{gem_name}.gemspec
-%{?scl:EOF}
+gem build ../%{gem_name}-%{version}.gemspec
 
 # %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
 # by default, so that we can move it into the buildroot in %%install
-%{?scl:scl enable %{scl} - << \EOF}
 %gem_install
-%{?scl:EOF}
 
 %install
 mkdir -p %{buildroot}%{gem_dir}
 cp -a .%{gem_dir}/* \
-        %{buildroot}%{gem_dir}/
-
-mkdir -p %{buildroot}%{gem_extdir_mri}/ruby_newt
-cp -a .%{gem_instdir}/ext/ruby_newt/*.so %{buildroot}%{gem_extdir_mri}/ruby_newt/
-touch %{buildroot}%{gem_extdir_mri}/gem.build_complete
-
-# Prevent dangling symlink in -debuginfo (rhbz#878863).
-rm -rf %{buildroot}%{gem_instdir}/{ext,tmp,.require_paths}
-
-%check
-%{?scl:scl enable %{scl} - << \EOF}
-%if 0%{?rhel} == 8
-# Workaround for: Ignoring newt-0.9.7 because its extensions are not built. Try: gem pristine newt --version 0.9.7
-mkdir gem_ext_test
-cp -a %{buildroot}%{gem_dir} gem_ext_test/
-mkdir -p gem_ext_test/gems/extensions/%{_arch}-%{_target_os}/$(ruby -r rbconfig -e 'print RbConfig::CONFIG["ruby_version"]')/
-cp -a %{buildroot}%{gem_extdir_mri} gem_ext_test/gems/extensions/%{_arch}-%{_target_os}/$(ruby -r rbconfig -e 'print RbConfig::CONFIG["ruby_version"]')/
-GEM_PATH="./gem_ext_test/gems:$GEM_PATH" ruby -e "require '%{gem_require_name}'"
-rm -rf gem_ext_test
-%else
-echo "Checking with %{buildroot}%{gem_dir}"
-GEM_PATH="%{buildroot}%{gem_dir}:$GEM_PATH" ruby -e "require '%{gem_require_name}'"
-%endif
-%{?scl:EOF}
```

Comment:
you removed the part that was building the native library, so it's not built anymore and thus not found during RPM file collection

Reply:
Oh, okay, I thought it was an "outdated" part of the installation. Will revert it back

Reply:
Oh it certainly is outdated, but something similar is required.

Here is what our template does today for gems that have extensions:
https://github.com/theforeman/foreman-packaging/blob/2d59deb9f5279e9c5537ea18b5c4185b8f9f9fb5/gem2rpm/default.spec.erb#L112-L122

 But it might not be sufficient -- ext building is not well standardized.

We have a check entry to try to catch errors:
https://github.com/theforeman/foreman-packaging/blob/2d59deb9f5279e9c5537ea18b5c4185b8f9f9fb5/gem2rpm/default.spec.erb#L131-L143